aiohttp/cookiejar.py

```diff
@@ -48,15 +47,16 @@ class CookieJar(AbstractCookieJar):
 
     DATE_YEAR_RE = re.compile(r"(\d{2,4})")
 
-    MAX_TIME = 2051215261.0  # so far in future (2035-01-01)
+    MAX_TIME: datetime.datetime = datetime.datetime.max.replace(
+        tzinfo=datetime.timezone.utc)
 
     def __init__(self, *, unsafe: bool=False) -> None:
         self._loop = get_running_loop()
         self._cookies = defaultdict(SimpleCookie)  #type: DefaultDict[str, SimpleCookie]  # noqa
         self._host_only_cookies = set()  # type: Set[Tuple[str, str]]
         self._unsafe = unsafe
-        self._next_expiration = ceil(self._loop.time())
-        self._expirations = {}  # type: Dict[Tuple[str, str], int]
+        self._next_expiration: datetime.datetime = self._next_whole_second()
+        self._expirations: Dict[Tuple[str, str], datetime.datetime] = {}
```

Comment:
Again, keep the hints in comments. 

```suggestion
        self._expirations = {}  # type: Dict[Tuple[str, str], datetime.datetime]
```

Reply:
Ok, I thought the plan was to deprecate 3.5, but will revert.